daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs

```diff
@@ -428,6 +442,136 @@ execClean projectOpts = do
             removeAndWarn "dist"
             putStrLn "Removed build artifacts."
 
+
+
+data Action = ACreate (LF.Qualified LF.TypeConName) 
+            | AExercise (LF.Qualified LF.TypeConName) LF.ChoiceName deriving (Eq, Ord, Show )
+
+
+startFromUpdate :: DS.Set (LF.Qualified LF.ExprValName) -> LF.World -> LF.Update -> DS.Set Action
+startFromUpdate seen world update = case update of 
+    LF.UPure _ e -> startFromExpr seen world e
+    LF.UBind (LF.Binding _ e1) e2 -> startFromExpr seen world e1 `Set.union` startFromExpr seen world e2
+    LF.UCreate tpl e -> Set.singleton (ACreate tpl) `Set.union` startFromExpr seen world e
+    LF.UExercise tpl chc e1 e2 e3 -> Set.singleton (AExercise tpl chc) `Set.union` startFromExpr seen world e1 `Set.union` maybe Set.empty (startFromExpr seen world) e2 `Set.union` startFromExpr seen world e3
+    LF.UFetch _ ctIdEx -> startFromExpr seen world ctIdEx
+    LF.UGetTime -> Set.empty
+    LF.UEmbedExpr _ upEx -> startFromExpr seen world upEx
+    LF.ULookupByKey _ -> Set.empty
```

Comment:
Is there a reason why you do not recurse here? If so, it would be nice to document this in a comment.

Reply:
Looks like a bug to me. But such bugs are inevitable if you don't use something automatic.